xarray/core/dataarray.py

```diff
@@ -103,8 +103,11 @@ def _remap_key(self, key):
         return indexing.remap_label_indexers(self.data_array, key)
 
     def __getitem__(self, key):
-        pos_indexers, new_indexes = self._remap_key(key)
-        return self.data_array[pos_indexers]._replace_indexes(new_indexes)
+        if not utils.is_dict_like(key):
```

Comment:
I just port `.loc[]` to `.sel()` here.

According to the discussions of [#974](https://github.com/pydata/xarray/issues/974#issue-171828347),
> Should we deprecate non-boolean indexing with [] and .loc[] and non-labelled arrays when some but not all dimensions are provided? Instead, we would require explicitly indexing like [key, ...] (yes, writing ...), which indicates "all trailing axes" like NumPy.

Do we need some warnings here?

And I think I don't yet well understand how `[key, ...]` works. 
Could someone provide an example use case of this?

Reply:
`...` means roughly "insert `:` as many times as necessary in this location so that there is one indexer per dimension".

I no longer think this sort of deprecation is a good idea -- this would be too surprising to NumPy users.